File: study/2022/10-W2/youjin.dart

class Solution {
  int findJudge(int n, List<List<int>> trust) {
    final List<int> trustCounts = List<int>.filled(n + 1, 0); // [0, 0, 0, ...];

    for (var trustProp in trust) {
      --trustCounts[trustProp.first];
      ++trustCounts[trustProp.last];
    }

    for (var i = 1; i < trustCounts.length; ++i) {
      if (trustCounts[i] == n - 1) return i;
    }

    return -1;
  }
}

// Runtime: 648 ms, faster than 66.67% of Dart online submissions for Find the Town Judge.
// Memory Usage: 177.9 MB, less than 33.33% of Dart online submissions for Find the Town Judge.

File: study/2022/10-W2/youjin.js
const findJudge = (n, trust) => {
  const trustCounts = new Array(n + 1).fill(0);
    
  for (let [i, j] of trust) {
    --trustCounts[i];
    ++trustCounts[j];
  }
    
  for (let i = 1; i < trustCounts.length; ++i) if (trustCounts[i] === n - 1) return i;
  return -1;
};

// Runtime: 102 ms, faster than 95.30% of JavaScript online submissions for Find the Town Judge.
// Memory Usage: 50 MB, less than 95.45% of JavaScript online submissions for Find the Town Judge.
